src/compiler/checker.ts

```diff
@@ -22469,9 +22469,11 @@ namespace ts {
 
             // If object literal is contextually typed by the implied type of a binding pattern, augment the result
             // type with those properties for which the binding pattern specifies a default value.
-            if (contextualTypeHasPattern) {
+            // If the object literal is spread into another object literal, skip this step and let the top-level object
+            // literal handle it instead.
+            if (contextualTypeHasPattern && !(node.parent.kind === SyntaxKind.SpreadAssignment && node.parent.parent.kind === SyntaxKind.ObjectLiteralExpression)) {

```

Comment:
can you ever have a SpreadAssignment whose parent isn't an ObjectLiteralExpression?

Reply:
Doh, good call. You cannot (validated by the `parent` property type on `ts.SpreadAssignment`). Updated.